app/certificates.go

```diff
@@ -0,0 +1,442 @@
+	}
+	dnsRoot := fmt.Sprintf("client.endentity.%s.%s", input.Organization, randomLetters)
+	now := time.Now()
+	conf := &x509.Certificate{
+		SerialNumber:          big.NewInt(2019),
+		Subject:               subject,
+		NotBefore:             now.Add(-time.Minute), // grace of 1 min
+		NotAfter:              now.Add(input.Duration),
+		BasicConstraintsValid: true,
+		DNSNames:              []string{dnsRoot},
+		MaxPathLen:            0,
+	}
+	var privateKey any
+	if isRSA {
+		privateKey, err = rsa.GenerateKey(rand.Reader, 4096)
+	} else {
+		privateKey, err = ecdsa.GenerateKey(elliptic.P384(), rand.Reader)
+	}
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to generate private key: %v", err)
+	}
+
+	var publicKey any
+	switch k := privateKey.(type) {
+	case *rsa.PrivateKey:
+		publicKey = &k.PublicKey
+	case *ecdsa.PrivateKey:
+		publicKey = &k.PublicKey
+	}
+	cert, err := x509.CreateCertificate(rand.Reader, conf, caCert, publicKey, caPrivateKey)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	certPEMBuffer := new(bytes.Buffer)
+	err = pem.Encode(certPEMBuffer, &pem.Block{
+		Type:  "CERTIFICATE",
+		Bytes: cert,
+	})
+	if err != nil {
+		return nil, nil, err
+	}
+
+	privBytes, err := x509.MarshalPKCS8PrivateKey(privateKey)
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to marshal private key: %v", err)
+	}
+	certPrivateKeyPEMBuffer := new(bytes.Buffer)
+	err = pem.Encode(certPrivateKeyPEMBuffer, &pem.Block{
+		Type:  "PRIVATE KEY",
+		Bytes: privBytes,
+	})
+	if err != nil {
+		return nil, nil, err
+	}
+	return certPEMBuffer.Bytes(), certPrivateKeyPEMBuffer.Bytes(), nil
+}
+
+func NewCertificatesCommand() (CommandOut, error) {
+	return CommandOut{
+		Command: &cli.Command{
+			Name:    "generate-certificates",
+			Aliases: []string{"gen"},
+			Usage:   "Generate tls certificates",
+			Subcommands: []*cli.Command{
+				{
+					Name:    "certificate-authority",
+					Usage:   "Generate a certificate authority",
+					Aliases: []string{"ca"},
+					Flags: []cli.Flag{
+						&cli.StringFlag{
+							Name:     "organization",
+							Usage:    "The name of the organization",
+							Aliases:  []string{"org"},
+							Required: true,
+						},
+						&cli.StringFlag{
+							Name:     "duration",
+							Usage:    "The duration of the ca certificate, for example: 1y30d (1 year and 30 days)",
+							Aliases:  []string{"d"},
+							Required: true,
+							Action: func(_ *cli.Context, v string) error {
+								d, err := utils.ParseDuration(v)
+								if err != nil {
+									return fmt.Errorf("failed to parse duration: %v", err)
+								}
+								if d > maxCADuration {
+									return fmt.Errorf("duration cannot be more than: %s", maxCADuration)
+								}
+								if d <= minCADuration {
+									return fmt.Errorf("duration cannot be less than: %s", minCADuration)
+								}
+								return nil
+							},
+						},
+						&cli.PathFlag{
+							Name:     CaCertificateFileFlagName,
```

Comment:
given these flags are for a command to generate ca certs, I think we can remove the word "ca/certificate authority" from them and just have it as --cert and --key

Reply:
ok I see the annoyance when it comes to client cert generation and avoiding flag name collisions there